src/App.js

```diff
@@ -1,7 +1,44 @@
+import React, { Component } from "react";
 import "./App.css";
+import Checkout from './Components/CheckOut'
+import Cart from './Components/Cart'
+import ItemList from "./Components/ItemList";
+import productData from './data/productData'
 
-const App = () => {
-  return <h1>Hello, world!</h1>;
+ class App extends React.Component {
+   constructor(props) {
+     super(props)
+     this.state = {
+       shoppingCart: [ ]
+     }
+     this.data = productData
+   }
+
+   addToCart = (item) => {
+     this.setState({ shoppingCart: [...this.state.shoppingCart, item]})
+   }
+ 
+
+  render () { 
+  const {shoppingCart} = this.state
+  const subTotal = shoppingCart.reduce((a, b ) => a + b.price, 0);
+  const tax = shoppingCart.reduce((a, b ) => a + b.price * 0.05)
```

Comment:
This line is throwing a type error when the shopping cart is empty.
You have to provide reduce with a second argument, a default/initial value, in case shopping cart is empty (the same way you passed 0 as a secong argument on line 24).